site/src/xServices/templateVersion/templateVersionXService.ts

```diff
@@ -0,0 +1,107 @@
+import { getFile, getTemplateVersionByName } from "api/api"
+import { TemplateVersion } from "api/typesGenerated"
+import { assign, createMachine } from "xstate"
+import untar from "js-untar"
+
+/**
+ * Content by filename
+ */
+export type TemplateVersionFiles = Record<string, string>
+
+/**
+ * File extensions to be displayed
+ */
+const allowedExtensions = ["tf", "md"]
+
+export const templateVersionMachine = createMachine(
+  {
+    predictableActionArguments: true,
+    id: "templateVersion",
+    schema: {
+      context: {} as {
+        orgId: string
+        versionName: string
+        version?: TemplateVersion
+        files?: TemplateVersionFiles
+        error?: unknown
+      },
+      services: {} as {
+        loadVersion: {
+          data: TemplateVersion
+        }
+        loadFiles: {
+          data: TemplateVersionFiles
+        }
+      },
+    },
+    tsTypes: {} as import("./templateVersionXService.typegen").Typegen0,
+    initial: "loadingVersion",
+    states: {
+      loadingVersion: {
+        invoke: {
+          src: "loadVersion",
+          onDone: {
+            target: "loadingFiles",
+            actions: ["assignVersion"],
+          },
+          onError: {
+            target: "done.error",
+            actions: ["assignError"],
+          },
+        },
+      },
+      loadingFiles: {
+        invoke: {
+          src: "loadFiles",
+          onDone: {
+            target: "done.ok",
+            actions: ["assignFiles"],
+          },
+          onError: {
+            target: "done.error",
+            actions: ["assignError"],
+          },
+        },
+      },
+      done: {
+        states: {
+          ok: { type: "final" },
+          error: { type: "final" },
+        },
+      },
+    },
+  },
+  {
+    actions: {
+      assignError: assign({
+        error: (_, { data }) => data,
```

Comment:
nice simple way to just show one error at a time (which is probably all that's possible in this case, but maybe we should do it more often)